modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaLayoutServicesCheck.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.ArrayUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Tamyris Bernardo
+ */
+public class UpgradeJavaLayoutServicesCheck extends BaseUpgradeCheck {
+
+	public String addMethodNewParameters(
+		int indexNewParameters, String methodCall, String newMethodCall,
+		List<String> newParameters, List<String> parameterList) {
+
+		parameterList.addAll(indexNewParameters, newParameters);
+
+		StringBundler sb = new StringBundler(7);
+
+		sb.append(getVariableName(methodCall));
+		sb.append(newMethodCall);
+		sb.append(StringPool.NEW_LINE);
+		sb.append(JavaSourceUtil.getIndent(methodCall));
+		sb.append(StringPool.TAB);
+		sb.append(StringUtil.merge(parameterList, StringPool.COMMA_AND_SPACE));
+		sb.append(StringPool.CLOSE_PARENTHESIS);
+
+		return sb.toString();
+	}
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		if (!fileName.endsWith(".java")) {
+			return content;
+		}
+
+		String newContent = content;
+
+		Matcher matcher = _addOrUpdateLayoutPattern.matcher(content);
+
+		while (matcher.find()) {
+			String methodCall = JavaSourceUtil.getMethodCall(
+				content, matcher.start());
+
+			if (!getVariableName(
+					methodCall
+			).contains(
+				"LayoutLocalServiceUtil"
```

Comment:
@TamyrisTorres, avoid using chaining. Please extract a variable.

And why do you didi this whey and the other you used the `hasClassOrVariableName` method?

Reply:
Because, the method call in this case is direct through the "LayoutLocalServiceUtil" class and not through the variable.